File: tests/omp/taskloop.cpp
void long_running_task();
void loop_body(int i, int j);

void parallel_work() {
   {
    long_running_task(); // can execute concurrently
    for (int i = 0; i < 10000; i++) { // can execute concurrently           
       for (int j = 0; j < i; j++) {
          loop_body(i, j);
       }
    }
   }
}

File: tests/omp/taskloop.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  !! Omp.taskgroup [tFirst; cFunDef "parallel_work";dBody];
  !! Omp.task [] [tBefore; cFun "long_running_task"];
  !! Omp.taskloop [Private ["j"]; Grainsize 500; Nogroup] [tBefore; cFor "i"];
)